domrl/engine/state_view.py

```diff
@@ -24,6 +24,20 @@ def __init__(self, player, is_player: bool):
         self.all_cards = [card for card in player.discard_pile] + \
             [card for card in player.hand] + \
             [card for card in player.draw_pile]
+        self.all_card_names = [card.name for card in player.discard_pile] + \
```

Comment:
I'm not a fan of setting this. I think the agent can define their own function to get these.